src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -223,8 +313,20 @@ impl WorkUnitStore {
       .flat_map(|id| workunit_records.get(id))
       .flat_map(|record| match record {
         WorkunitRecord::Completed(_) => None,
+        WorkunitRecord::Started(StartedWorkUnit { metadata, .. }) if !metadata.display => None,
         WorkunitRecord::Started(c) => Some(c.clone()),
       })
+      .map(|started: StartedWorkUnit| {
+        if let Some(ref parent_id) = started.parent_id {
+          if !should_display_workunit(workunit_records, parent_id) {
```

Comment:
This is a bit odd. We're using the "display" value of the parent to decide whether a node should be displayed?

Reply:
This is using the display value of the parent to decide if the parent ID should be included on the record as passed to python. This is necessary because otherwise we would sometimes show a parent id value in a workunit that has no displayed parent, and there was an existing test that checked for this. We could change this behavior, of course, but I think it does make sense not to show a parent ID that won't otherwise be displayed in workunit output.

Reply:
Interesting. That won't work "generally", because it might be your parent's parent that is the next display node, so this should probably walk until it finds another "display" parent.

Reply:
@stuhood pushed a fix for this